bin/circle/search-circle-builds

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env ruby
+# A script to analyze a range of CircleCI builds for known failures.
+# For available options, run ./search-circle-builds -h
+
+require 'pp'
+require 'json'
+require 'memoist'
+require 'open-uri'
+require 'optparse'
+
+CIRCLE_API_BASE = 'https://circleci.com/api/v1.1/project/github/code-dot-org/code-dot-org'
+
+KNOWN_FAIL_TYPES = [
+    {
+        search_string: 'ConnectionTimeoutError',
+        step_string: 'run_tests',
+        container: 1
+    },
+    {
+        search_string: 'io_fillbuf',
+        step_string: 'run_tests',
+        container: 1
+    },
+]
```

Comment:
Good call re: optional grep parameter, that was initially why I set out to write the tool but then was thinking it'd be handy to scan a range and get a sense of failure categorization.

I'll add the base64 checker and a grep option as well.